spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -792,13 +794,14 @@ private static class Context implements Serializable {
         Schema dataSchema,
         SparkWriteConf writeConf,
         LogicalWriteInfo info,
-        SparkWriteRequirements writeRequirements) {
+        SparkWriteRequirements writeRequirements,
+        int formatVersion) {
       this.dataSchema = dataSchema;
       this.dataSparkType = info.schema();
       this.dataFileFormat = writeConf.dataFileFormat();
       this.targetDataFileSize = writeConf.targetDataFileSize();
       this.deleteSparkType = info.rowIdSchema().get();
-      this.deleteFileFormat = writeConf.deleteFileFormat();
+      this.deleteFileFormat = formatVersion >= 3 ? FileFormat.PUFFIN : writeConf.deleteFileFormat();
```

Comment:
Isn't this redundant with the `useDVs` below? When going for `PUFFIN`, it implies that it uses DVs.